### DISCUSSION TITLE
Is it possible to read matcher data from external sources?

### DISCUSSION THREAD

**Discussion by hktalent** [projectdiscovery/nuclei#1842] [Q&A] [answered]
how use db or json files in yaml?
How do I only write one yaml, how do I connect to the local db, or add several json and other rule files for fingerprint recognition, or other recognition activities

thanks

**Comment by ehsandeep** [comment]
@hktalent could you elaborate more about this question/feature?

Are you looking to load templates files from DB or JSON files instead of YAML?

**Comment by hktalent** [comment]
@ehsandeep I hope that YAML can use database or json file data, because I use yaml for fingerprint identification, and a large amount of fingerprint identification data is stored in the database

**Comment by hktalent** [comment]
@ehsandeep Scenes:
In order to put some scanning rules in the database, or in the json file
At this time, you need to use them in yaml, but because you don't want to see all these data embedded in yaml, it is inconvenient to maintain, and the yaml file will become very large

how use https://github.com/wappalyzer/wappalyzer json in yaml

**Comment by forgedhallpass** [comment, marked as answer]
@hktalent 

Reading matchers from databases is not supported, nor we intend to add such functionality at this point in time. JSON files would be even more verbose than YAML.

In the future we might implement [this](https://github.com/projectdiscovery/nuclei/issues/1510) enhancement, which will enable reading and re-using matcher data in a more convenient way.

With regards to `wappalyzer`, we already have a template called [`tech-detect.yaml`](https://github.com/projectdiscovery/nuclei-templates/blob/master/technologies/tech-detect.yaml). A new `-automatic-scan` flag has also been added that executes relevant templates only.

**Reply by hktalent** [reply]
thanks
